### PR TITLE
Allow scheduler to enqueue finalize jobs tasks when marking stale job as incomplete

### PR DESCRIPTION
**Pull request by Martchus**
See https://progress.opensuse.org/issues/71386

### CODE REVIEW

File: t/20-stale-job-detection.t [hunk 1]
```diff
@@ -22,19 +22,22 @@ use lib "$FindBin::Bin/lib";
...
 
 my $schema = OpenQA::Test::Database->new->create(fixtures_glob => '01-jobs.pl 02-workers.pl 06-job_dependencies.pl');
 my $jobs   = $schema->resultset('Jobs');
 $jobs->find(99963)->update({assigned_worker_id => 1});
 $jobs->find(99961)->update({assigned_worker_id => 2});
 $jobs->find(80000)->update({state              => ASSIGNED, result => NONE, assigned_worker_id => 1});
 
-setup_mojo_app_with_default_worker_timeout;
+OpenQA::App->set_singleton(my $app = OpenQA::Scheduler->new);
+$app->_setup;
```
Comment (kraih) [inline t/20-stale-job-detection.t:39]: This probably shouldn't be an underscore method if it's used as part of the public API.

**Comment by kraih** [comment]
Seems like a good idea.

File: t/20-stale-job-detection.t (hunk 1 above)
Comment (Martchus) [reply t/20-stale-job-detection.t:39]: This is just called from a test.

File: t/20-stale-job-detection.t [hunk 2]
```diff
@@ -22,19 +22,22 @@ use lib "$FindBin::Bin/lib";
...
-use OpenQA::Test::Utils qw(setup_mojo_app_with_default_worker_timeout redirect_output);
+use OpenQA::Test::Utils qw(redirect_output);
 
 my $schema = OpenQA::Test::Database->new->create(fixtures_glob => '01-jobs.pl 02-workers.pl 06-job_dependencies.pl');
 my $jobs   = $schema->resultset('Jobs');
 $jobs->find(99963)->update({assigned_worker_id => 1});
 $jobs->find(99961)->update({assigned_worker_id => 2});
 $jobs->find(80000)->update({state              => ASSIGNED, result => NONE, assigned_worker_id => 1});
 
-setup_mojo_app_with_default_worker_timeout;
```
Comment (okurz) [inline t/20-stale-job-detection.t:37]: When you not use this function anymore, is it still used somewhere else? If not, delete, if only once better move the function to the single place where used

File: t/20-stale-job-detection.t [hunk 3]
```diff
@@ -22,19 +22,22 @@ use lib "$FindBin::Bin/lib";
...
+OpenQA::App->set_singleton(my $app = OpenQA::Scheduler->new);
+$app->setup;
+$app->log(undef);
```
Comment (okurz) [inline t/20-stale-job-detection.t:40]: Can you please describe if this has as significant runtime impact on the test?

**Review by okurz — COMMENTED** [review]
Didn't you come to this idea because there is the possibility in the scheduler that the gru unavailability was silently ignored? Better remove the according `... if gru` then

File: t/20-stale-job-detection.t (hunk 2 above)
Comment (Martchus) [reply t/20-stale-job-detection.t:37]: It is still used on multiple places.

File: t/20-stale-job-detection.t (hunk 3 above)
Comment (Martchus) [reply t/20-stale-job-detection.t:40]: I wanted to keep the test as simple as possible using the default app. Instantiating the scheduler app instead takes definitely longer but I couldn't measure a notable difference. (This has no further sleeps, db initializations if that's your concern.)

**Comment by Martchus** [comment]
> Didn't you come to this idea because there is the possibility in the scheduler that the gru unavailability was silently ignored? Better remove the according ... if gru then

That's still required for other tests and I don't think it is worth loading the Gru plugin in all the tests. Besides, I think we should be able to mark jobs as done even when the Gru plugin could not be loaded. These finalize jobs are really an optional performance optimization and could be enqueued anytime later if needed.

**Review by okurz — APPROVED** [review]

**Review by kraih — APPROVED** [review]

**Pull request merged**